learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -211,7 +241,37 @@ async def _process_student_documents(student):
     # TODO saved skipped doc ids somewhere?
 
 
-async def _google(request):
+async def _handle_canvas_authorization(request):
+    '''
+    Handle Canvas authorization
+    '''
+    if 'error' in request.query:
+        return {}
+
+    token_uri = settings.pmss_settings.token_uri(types=['lms', 'canvas_oauth'])
+    url = token_uri
+
+    params = {
+        "grant_type": "refresh_token",
+        'client_id': settings.pmss_settings.client_id(types=['lms', 'canvas_oauth']),
+        'client_secret': settings.pmss_settings.client_secret(types=['lms', 'canvas_oauth']),
+        "refresh_token": settings.pmss_settings.refresh_token(types=['lms', 'canvas_oauth'])
+    }
+    async with aiohttp.ClientSession(loop=request.app.loop) as client:
+        async with client.post(url, data=params) as resp:
+            data = await resp.json()
+        assert 'access_token' in data, data
+
+        # get user profile
+        canvas_headers = {'Authorization': 'Bearer ' + data['access_token']}
```

Comment:
We ought to be storing more than just the `access_token` for each user. We should include the `refresh_token` if available and some additional metadata about when the sign-in occurred.